pkg/apis/kubeone/v1beta3/types.go

```diff
@@ -440,6 +440,10 @@ type KubevirtSpec struct {
 	// InfraNamespace is the namespace that KubeVirt provider will use to create and manage resources in the infra cluster,
 	// such as VirtualMachines, VirtualMachineInstances, etc...
 	InfraNamespace string `json:"infraNamespace,omitempty"`
+	// InfraClusterKubeconfig represents the kubeconfig that points to KubeVirt infra cluster(a cluster with kubevirt is installed).
+	// This Kubeconfig will be used to create and manage KubeVirt specific resources such as DataVolumes. The value must be
+	// set in a base64 format.
+	InfraClusterKubeconfig string `json:"infraClusterKubeconfig,omitempty"`
```

Comment:
Please add validation to ensure this is base64-encoded. Is this kubeconfig required?

Reply:
Why? This is something that KubeOne doesn't care about and it is part of the CSI implementation. Thus the CSI driver itself will fail not KubeOne roll out.

Reply:
If you provide plain-text kubeconfig here, creating the `kubevirt-csi-driver` Secret would fail. Because of that, KubeOne would fail as well, i.e. it wouldn't fully provision or upgrade a cluster. We handle such cases by adding validation rules so that we catch cases for which we know that will result in a failure in later steps.

Reply:
I got you. But in that case would actually KubeOne fail? If yes then sure I can add a validation no problem.

Reply:
Yes, that's correct, KubeOne would fail in this case. Thanks a lot for adding this validation!